File: LeetCodeDSA/860.py
from collections import Counter, defaultdict
def lemonadeChange(bills):
    cnt = defaultdict()
    cnt[5] = 0 
    cnt[10] = 0
    cnt[20] = 0
    for i in bills:
        if i == 5: 
            cnt[5] += 1
        else: 
            if i == 10: 
                cnt[5] -= 1
                cnt[10] += 1
                if cnt[5] < 0: 
                    return False
            if i == 20: 
                if cnt[10] > 0 and cnt[5] > 0: 
                    cnt[10] -=1
                    cnt[5] -= 1
                else: 
                    cnt[5] -= 3
                    if cnt[5] < 0 : 
                        return False
    return True
    
    
print(lemonadeChange(bills = [5,5,10,10,20]))

File: LeetCodeDSA/860.c
#include <stdio.h>
#include <stdbool.h>

bool lemonadeChange(int* bills, int billsSize){
    int fiveDollarBill = 0; 
    int tenDollarBill = 0; 
    
    for(int i = 0; i < billsSize; i++){
        if(bills[i] == 5){
            fiveDollarBill++; 
        }else{ 
            if(bills[i] == 10){
                if(fiveDollarBill - 1 < 0){
                    return false;
                }else{
                    fiveDollarBill--; 
                    tenDollarBill++;
                }
            }
            if(bills[i] == 20){ 
                if(fiveDollarBill > 0 && tenDollarBill > 0){
                    fiveDollarBill--; 
                    tenDollarBill--; 
                }else{
                    fiveDollarBill -= 3;
                    if(fiveDollarBill < 0){
                        return false;
                    }
                }
            }
        }
    }
    return true;
}

int main(){
    int billsSize = 5;
    int bills[] = {5,5,5,10,20}; 
    if(lemonadeChange(bills, billsSize)){
        printf("True");
    }else{
        printf("False");
    }
    return 0;
}